python/src/uagents/communication.py

```diff
@@ -163,10 +163,12 @@ async def send_exchange_envelope(
                     success = resp.status == 200
                     if success:
                         if sync:
-                            # If the message is synchronous but not verified, return the envelope
                             env = Envelope.model_validate(await resp.json())
-                            if env.signature is None:
-                                return env
+                            if env.signature and not env.verify():
```

Comment:
If verification fails (`env.verify()`) it'll raise an Exception (ValueError, BadSignatureError or BadDigestError) so we need to catch that as the only bool ever to be returned is `True`